src/core/server/legacy/logging/legacy_logging_server.ts

```diff
@@ -90,9 +101,16 @@ export class LegacyLoggingServer {
   }
 
   public log({ level, context, message, error, timestamp, meta = {} }: LogRecord) {
+    const { tags = [], ...metadata } = meta;
+    const data = error
+      ? error
+      : isEmptyObject(metadata)
+      ? message
+      : attachMetaData(message, metadata);
```

Comment:
Nested ternary is a little hard to read. There's a few options you could use, but it could probably just be:

```ts
const data = error || attachMetadata(message, metadata);
```

And then have `attachMetadata` handle the empty object case, which would make the function more defensively written anyways.

Reply:
I'd decided not to force metadata format here. There are lots of mutually exclusive conditions. So if we hit this branch https://github.com/restrry/kibana/blob/97f4567ae9d912b1ec02942328bffe86a9d12b3b/src/legacy/server/logging/log_format.js#L162, we skip others. I'd say it's too easy to shoot ourselves in the foot.

Reply:
switched to a separate function